### PR TITLE
LPD-15853 Get the long value of the relationship field before checking if it is null

### PR DESCRIPTION
**Pull request by ccorreagg**
Related ticket: https://liferay.atlassian.net/browse/LPD-15853

---

How to reproduce it: Check the related ticket.

The purpose of this ticket is to fix the current validation that is being done over the relationship field content.

In the related ticket example, given the curl:

```
curl -X 'POST'  \
     -u 'test@liferay.com:test' \
     'http://localhost:8080/o/c/students' \
     -H 'Content-type: application/json' \
     -d '{
         "studentName":"Carlos",
        "r_universityStudents_c_universityId": 0
     }'
```

The value `0` is being parsed as an `Integer`.

In order to validate the null value, the `Validator.isNull` method is used [here](https://github.com/liferay/liferay-portal/blob/ca882b90a970d51cc644e0eaf51cbb05f7411532/modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/RelationshipObjectFieldBusinessType.java#L96). With the current implementation of the method, if the argument is an instance of the `Integer` class, then the isNull method return `false`, as it is not covered by any if clause. That triggers a following query over the ObjectEntry service with the id 0, that will throw an exception.

We have tried to update the implementation of the `isNull` value, but it would be a breaking change, as an integer 0 value is not being considered as null value, so the components that are using it should be updated too, that is why I think it is more safe to keep the current implementation of the `isNull` method.

With the code of this PR, we are parsing the current value as a Long, as it is the one that must be.

### CODE REVIEW

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
ci:test:relevant

File: portal-kernel/src/com/liferay/portal/kernel/util/Validator.java [hunk 1]
```diff
@@ -897,6 +897,11 @@ public static boolean isNull(Object object) {
 		if (object instanceof Long) {
 			return isNull((Long)object);
 		}
+		else if (object instanceof Number) {
+			Number number = (Number)object;
+
+			return isNull(number.longValue());
```
Comment (4lejandrito) [inline portal-kernel/src/com/liferay/portal/kernel/util/Validator.java:903]: What if the number is a float? would this still work?

File: portal-kernel/src/com/liferay/portal/kernel/util/Validator.java [hunk 2]
```diff
@@ -897,6 +897,11 @@ public static boolean isNull(Object object) {
 		if (object instanceof Long) {
```
Comment (4lejandrito) [inline portal-kernel/src/com/liferay/portal/kernel/util/Validator.java:897]: Is this now redundand? the number case should cover it, right?

**Comment by ccorreagg** [comment]
I am updating the code just to support Integer, as this method could have being called with any kind of object, so the best to ensure we don't break anything is to add support to Integer

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:relevant

**Review by 4lejandrito — APPROVED** [review]

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:forward

**Comment by ccorreagg** [comment]
ci:forward

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:stop

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:relevant

**Review by 4lejandrito — APPROVED** [review]
Much safer this way :+1:.

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
ci:forward:force

**Pull request closed without merging**